src/account.ts

```diff
@@ -388,15 +547,19 @@ export class ArgentAccount extends Account {
             );
         }
 
-        return new ArgentAccount(contract, privateKey, publicKey, keyPair, "0", "0", null);
+        return new ArgentAccount(contract, privateKey, publicKey, keyPair, "0", "0", null, hre);
     }
 
-    getExecutionFunctionName(): string {
+    protected getExecutionFunctionName(): string {
         return "__execute__";
     }
 
-    async getNonce(): Promise<bigint> {
+    protected async getNonce(): Promise<bigint> {
         const { nonce } = await this.starknetContract.call("get_nonce");
         return nonce;
     }
+
+    protected hasRawOutput(): boolean {
```

Comment:
Why is this a function?

Reply:
If you're asking why it isn't a property, it doesn't have to be, it's just following the same paradigm as `getExecutionFunctionName`, which could also be a property set through constructor.